From 3925d8e0233563648f7b234e0638ac4cb291feb6 Mon Sep 17 00:00:00 2001
From: Patrick Gaskin <patrick@pgaskin.net>
Date: Sun, 24 May 2026 23:34:19 -0400
Subject: [PATCH] Partially revert 428ad443d30c0808429f2f4e81cebd7045851b70

---
 fncompiler.go | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/fncompiler.go b/fncompiler.go
index 8a3bbdc..538c9d2 100644
--- a/fncompiler.go
+++ b/fncompiler.go
@@ -225,13 +225,17 @@ func (fn *funcCompiler) pushPureIf(pure bool, expr ast.Expr) {
 	}
 }
 
-// Flushes the stack before pushing expr to the value stack.
+// Pushes the materialization of expr to the value stack.
 func (fn *funcCompiler) push(expr ast.Expr) {
 	if fn.blocks.top().unreachable {
 		return
 	}
-	fn.flush()
-	fn.pushPure(expr)
+	tmp := fn.newTempVal()
+	fn.emit(&ast.AssignStmt{
+		Tok: token.DEFINE,
+		Lhs: []ast.Expr{tmp},
+		Rhs: []ast.Expr{expr}})
+	fn.pushConst(tmp)
 }
 
 // Drops a value from the value stack.
-- 
2.54.0

